state/validation.go

```diff
@@ -123,11 +123,9 @@ func validateBlock(evidencePool EvidencePool, stateDB dbm.DB, state State, block
 	}
 
 	// Limit the amount of evidence
-	maxNumEvidence, _ := types.MaxEvidencePerBlock(state.ConsensusParams.Block.MaxBytes)
-	numEvidence := int64(len(block.Evidence.Evidence))
-	if numEvidence > maxNumEvidence {
-		return types.NewErrEvidenceOverflow(maxNumEvidence, numEvidence)
-
+	numEvidence := len(block.Evidence.Evidence)
+	if numEvidence > int(state.ConsensusParams.Evidence.MaxNumEvidence) {
```

Comment:
```suggestion
	// MaxNumEvidence is capped at uint16, so conversion is always safe.
	if numEvidence > int(state.ConsensusParams.Evidence.MaxNumEvidence) {
```

Reply:
Do you also want me to add that comment to the params.go file as well

Reply:
no